### ISSUE TITLE
Overlay hit detection reports hits on obscured elements

### ISSUE THREAD

**Issue by tvolkert** [flutter/flutter#3819]
By default, iOS accessibility hit detection will traverse the entire a11y element tree*, stopping only when it finds an element for which the following two conditions are both met:
1. The `x` and `y` coordinates of the hit test overlap the element's bounding box
2. The element responds to `isAccessibilityElement` with `YES`

\* _Note that because of this algorithm, when building the a11y element tree for iOS, we reverse the order of the real element's children to ensure that elements that are painted last are traversed first for the purposes of accessibility hit detection._

A strange consequence to this algorithm is that by default, elements that are _visually_ covered by an overlay are still exposed to accessibility hit detection.  Imagine the following simplistic element tree:
- `[HiddenContainer @0,0+400x600]`
  - `[HiddenButton @100,100+200x50]`
- `[OverlayContainer @0,0+400x600]`
  - `[OverlayButton @100,400+200x50]`

In this case, `OverlayContainer` completely obscures `HiddenContainer`, so `HiddenButton` is not visible and should not respond to touch.  But given the accessibility hit detection rules above, we're faced with two bad choices:
1. Report `[OverlayContainer isAccessibilityElement] == NO` so that VoiceOver will correctly detect hits on `OverlayButton`.  But in this case, it will still report a hit on `HiddenButton` at coordinates `200,150` (because it didn't hit `OverlayButton`, so it kept traversing the tree).
2. Report `[OverlayContainer isAccessibilityElement] == YES` to avoid the condition above.  But in this case, coordinates `200,450` will report a hit on `OverlayContainer`, **not** `OverlayButton`

We may have to intelligently massage the element tree to account for this, such that the accessibility element tree does not have a 1:1 relationship with the real element tree.  Since we're already reversing the child order of elements, there's already precedent for doing this king of thing.

**Comment by tvolkert** [comment]
It's also possible that we could use #3821 to fix this...

**Comment by tvolkert** [comment]
@Hixie this is a more detailed description of what we were talking about yesterday.

**Comment by Hixie** [comment]
See also https://github.com/flutter/flutter/issues/1705

**Comment by goderbauer** [comment]
This is fixed. Unreachable nodes are excluded from the tree with the `BlockSemantics` widget.
